src/Data/Primitive/Vec.hs

```diff
@@ -93,6 +135,14 @@ instance (Show a, Prim a, KnownNat n) => Show (Vec n a) where
           . group . encloseSep (flatAlt "< " "<") (flatAlt " >" ">") ", "
           . map viaShow
 
+vecOfList :: forall n a. (KnownNat n, Prim a) => [a] -> Vec n a
+vecOfList vs = runST $ do
+  let n :: Int = fromIntegral $ natVal $ Proxy @n
+  mba <- newByteArray (n * sizeOf (undefined :: a))
+  zipWithM_ (writeByteArray mba) [0..n] vs
```

Comment:
n-1 ?

Reply:
absolutely, great catch!

Reply:
actually, this doesn't matter. You could use `[0..]` and `zipWith` will stop when it reaches the end of `vs`